pkg/cmd/roachtest/tests/backup.go

```diff
@@ -164,58 +171,338 @@ func registerBackupNodeShutdown(r registry.Registry) {
 
 }
 
+// removeJobClaimsForNodes nullifies the `claim_session_id` for the job
+// corresponding to jobID, if it has been incorrectly claimed by one of the
+// `nodes`.
+func removeJobClaimsForNodes(
+	ctx context.Context, t test.Test, db *gosql.DB, nodes option.NodeListOption, jobID jobspb.JobID,
+) {
+	if len(nodes) == 0 {
+		return
+	}
+
+	n := make([]string, 0)
+	for _, node := range nodes {
+		n = append(n, strconv.Itoa(node))
+	}
+	nodesStr := strings.Join(n, ",")
+
+	removeClaimQuery := `
+UPDATE system.jobs
+   SET claim_session_id = NULL
+WHERE claim_instance_id IN (%s)
+AND id = $1
+`
+	_, err := db.ExecContext(ctx, fmt.Sprintf(removeClaimQuery, nodesStr), jobID)
+	require.NoError(t, err)
+}
+
+// waitForJobToHaveStatus waits for the job with jobID to reach the
+// expectedStatus.
+func waitForJobToHaveStatus(
+	ctx context.Context,
+	t test.Test,
+	db *gosql.DB,
+	jobID jobspb.JobID,
+	expectedStatus jobs.Status,
+	nodesWithAdoptionDisabled option.NodeListOption,
+) {
+	if err := retry.ForDuration(time.Minute*2, func() error {
```

Comment:
nit: I might prefer to define func above, so as to have one less layer of indentation. Non-blocking, and possibly not even right :)